modos/codes.py

```diff
@@ -0,0 +1,68 @@
+"""Utilities to automatically find / recommend terminology codes from text."""
+from dataclasses import dataclass
+import requests
+from typing import Optional, Protocol
+
+
+SLOT_TERMINOLOGIES = {
+    "cell_type": "https://purl.obolibrary.org/obo/cl.owl",
+    "source_material": "https://purl.obolibrary.org/obo/uberon.owl",
+    "taxon_id": "https://purl.obolibrary.org/obo/ncbitaxon/subsets/taxslim.owl",
+}
+
+
+@dataclass
+class Code:
+    label: str
+    uri: str
+
+
+class CodeMatcher(Protocol):
+    def find_codes(self, query: str) -> list[Code]:
+        ...
+
+
+class LocalCodeMatcher(CodeMatcher):
+    """Find ontology codes for a given text by running a local term matcher."""
+
+    def __init__(self, slot: str):
+        self.slot = slot
+        try:
+            from pyfuzon import TermMatcher
+
+            self.matcher = TermMatcher.from_files([SLOT_TERMINOLOGIES[slot]])
+        except ImportError:
+            raise ValueError(
+                """No endpoint provided and pyfuzon not installed,
+                cannot do code matching."""
+            )
+
+    def find_codes(self, query: str) -> list[Code]:
+        return self.matcher.top(query, 50)
```

Comment:
**suggestion(enhancement, non-blocking)**: we're hardcoding top50 matches if I get it correctly: any chance we can parametrize it somewhere? Can be a separate issue for another PR

Reply:
good point, this is parametrized as of ac78fa1